test/src/mistralai_client_dart_base_chat_stream_test.dart

```diff
@@ -0,0 +1,251 @@
+import 'package:mistralai_client_dart/mistralai_client_dart.dart';
+import 'package:test/test.dart';
+
+import 'fakes.dart';
+import 'fixtures.dart';
+
+void main() {
+  group('MistralAIClient chat stream test', () {
+    test(
+        'given response chunks '
+        'when calling chat stream '
+        'then should result in chat completion chunk results emissions', () {
+      // given
+      final mockHttpClient = FakeStreamedResponseHttpClient(
+        responseChunks: [
+          chatCompletionChunkResponse,
+          'some random chunk which should be ignored',
+          chatCompletionChunkResponse,
+          'another random chunk which should be ignored',
+          chatCompletionChunkResponse,
+        ],
+      );
+      final mistralClient = mistralAIClientOf(client: mockHttpClient);
+
+      // when
+      final result = mistralClient.chatStream(chatParamsOf());
+
+      // then
+      expect(
+        result,
+        emits(isA<ChatCompletionChunkResult>()),
+      );
```

Comment:
Shouldn't we check that all 3 results are emitted? 🤔

Reply:
I think the main point of this test is to check if the proper type was emitted, maybe if we refactor the test name it will make more sense. WDYT?
```
  test(
        'given response chunks '
        'when calling chat stream '
        'then should result in ChatCompletionChunkResult emission', () {
      // given
      final mockHttpClient = FakeStreamedResponseHttpClient(
        responseChunks: [
          chatCompletionChunkResponse,
          'some random chunk which should be ignored',
          chatCompletionChunkResponse,
          'another random chunk which should be ignored',
          chatCompletionChunkResponse,
        ],
      );
      final mistralClient = mistralAIClientOf(httpClient: mockHttpClient);

      // when
      final result = mistralClient.chatStream(chatParamsOf());

      // then
      expect(
        result,
        emits(isA<ChatCompletionChunkResult>()),
      );
    });
```

Reply:
I thought that `emits` checks against all events but that's not true. So to fix it we need to use `emitsInOrder` and add isA<ChatCompletionChunkResult>() 3 times in array.

Reply:
```suggestion
      expect(
        result,
        emitsInOrder([
          isA<ChatCompletionChunkResult>(),
          isA<ChatCompletionChunkResult>(),
          isA<ChatCompletionChunkResult>(),
        ]),
      );
```